backend/src/controllers/profileController.ts

```diff
@@ -0,0 +1,79 @@
+import { v4 } from "uuid";
+import { createProfileSchema, getProfileSchema } from "../helpers/schemas";
+import dbPool from "../db/index";
+import { NextFunction, Request, Response } from "express";
+
+interface User {
+  id: string;
+  name: string;
+  email: string;
+  hobby: string;
+}
+
+class ProfileRepository {
+  pool = dbPool.getPool();
+
+  async getAllProfiles() {
+    return this.pool.query("SELECT * from profile");
+  }
+
+  async getProfileById(userId: string) {
+    return this.pool.query("SELECT * from profile where uuid = $1", [userId]);
+  }
+
+  async createProfile(user: User) {
+    return this.pool.query(
+      "INSERT INTO profile (uuid, username, email, hobby, created_on) VALUES ($1, $2, $3, $4, NOW())",
+      [user.id, user.name, user.email, user.hobby]
+    );
+  }
+}
+
+const profileRepository = new ProfileRepository();
```

Comment:
You need to instantiate this controller in a constructor (also has to be private readonly )